content/blog/experiences-with-ci-cd.pt-br.md

```diff
@@ -0,0 +1,476 @@
+
+Commits que chegavam à `master`, usada para produção no nosso caso, chegariam
+automaticamente ao ambiente de produção --- a versão do site servida pelo
+servidor. Nesse sentido, isso foi um exemplo de deployment contínuo, mas _se não
+há CI, pode haver CD?_ É necessária a presença dessa "fase de teste automático"
+para que o deployment seja contínuo?
+
+Eu diria que não. CI e CD são ambos premissas fundamentais da cultura DevOps,
+mas a implantação dos dois costuma ser feita separadamente. Fossem eles
+genuinamente inseparáveis, provavelmente teríamos uma expressão só a designar a
+"unidade" composta por ambos.
+
+[techtarget-cd]: https://www.techtarget.com/searchitoperations/definition/continuous-deployment
+
+## O caminho para CI/CD propriamente dito
+
+Meus olhos se abriram para CI/CD quando comecei a trabalhar. Até então, mal
+sabia o que significavam, se é que havia realmente ouvido falar dos dois. A mim,
+CI era a segunda parte de [Travis CI][travis-ci], uma ferramenta que rodava
+builds no repositório do XMonad de forma automática, e CD era um formato de
+mídia ótica.
+
+Começo a trabalhar e quase que de imediato, me mostram o [Jenkins][jenkins]. Me
+foi apresentando como a ferramenta de escolha para a maioria dos desenvolvedores
+para criar esteiras de CI/CD, e dá para ver por quê. No entanto, o sentimento
+que o Jenkins passa, quanto mais experiência com ele você adquire, é de um
+programa equivalente a um pano de prato feito de vários remendos e que consome
+mais recursos do que deve ser necessário.
+
+Inexperiente e com os olhos brilhando, refletindo uma nova admiração pela
+construção de esteiras de desenvolvimento, busquei escrever algumas
+`Jenkinsfile`s para automatizar processos como alterações no firmware do meu
+eReader.
+
+Criados e configurados repositórios aqui e ali, lendo a documentação enquanto
+pensava "rapaz, que trabalheira isso deu, mas vai valer a pena", minha atenção
+se voltou a [GitHub Actions][gh-actions]. Só algumas linhas de YAML e um pouco
+de documentação e um workflow estava pronto, o segundo já em progresso. O GitHub
+lida com tudo nativamente, e até roda os workflows de graça contanto que o
+repositório continuasse público, ou eu não tivesse ultrapassado o limite de
+minutos do mês.
+
+[gh-actions]: https://github.com/actions
+[travis-ci]: https://travis-ci.com
+[jenkins]: https://jenkins.io
+
+### O motivo de eu nem sempre querer usar Jenkins
+
+Aqui vai um exemplo encurtado de uma `Jenkinsfile` comum, usando a sintaxe de
+pipeline declarativa:
+
+```groovy
+pipeline {
+  agent      { /* ... */ }
+  options    { /* ... */ }
+  parameters { /* ... */ }
+  stages {
+    stage('faz alguma coisa') {
+      steps {
+        agoraSimFazAlgumaCoisa
+      }
+    }
+  }
+}
+```
+
+As opções de build (`options`) costumam ocupar múltiplas linhas, bem como a
+specificação de `agent` e os `parameters`, no caso de uma pipeline
+parametrizada. Em cima disso, uma pipeline com um único estágio é bem raro,
```

Comment:
"algo bem raro" or "é bem rara"

Reply:
Went with the former, since it's difficult to "genderize" nouns that come from a non-gendered language like English. Hence we say "minha branch" and "meu branch" somewhat interchangeably. Applied in ead6a2d.